### PR TITLE
feat: release script

### PR DESCRIPTION
**Pull request by asvae**
* See [notion](https://www.notion.so/epicmax/f3c7ff0a60bb46568ff1b2db55a58880#37051b51fcc64837919debdf49f4ea87) for short explanation.
* See [docs](https://github.com/epicmaxco/vuestic-ui/pull/1986/files#diff-1b4f2143cc29c48285744bcdcc8683b18b3ff5248666adf936d6be14ff6678a0) for longer explanation.

## Next steps
* #1987
* #1988
* #1989

## Types of changes
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,10 @@
     "generate:docs": "yarn workspace docs generate:docspage",
     "generate:component": "yarn workspace docs generate:component",
     "run-all": "wt -d packages/ui --title vuestic-book cmd /k yarn serve --port 8080; split-pane -d packages/docs --title vuestic-docs cmd /k yarn serve --port 8081",
-    "version:bump-patch": "cd packages/ui && ts-node build/bump-version.ts --type patch && build && npm publish --tag=next --dry-run"
+    "release:montly": "cd packages/ui && tsx build/bump-version.ts --type=monthly",
```
Comment (m0ksem) [inline package.json:34]:
```suggestion
    "release:montly": "cd packages/ui && tsx build/bump-version.ts --type=monthly && cd ../..",
```

or move it to packages/ui and use `yarn workspace vuestic-ui release`

**Review by m0ksem — APPROVED** [review]

File: packages/ui/scripts/release-script.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+const argv: {
+  releaseType: ReleaseType,
+  semverReleaseType: string,
+  dryRun: boolean
+} = yargs(hideBin(process.argv))
+  .option('releaseType', {
+    description: 'Internal release type',
+    choices: ['monthly', 'weekly', 'next', 'experimental'],
+    default: 'alpha',
```
Comment (m0ksem) [inline packages/ui/scripts/release-script.ts:35]: Alpha is not a correct type

File: packages/ui/scripts/release-script.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  // 1.2.3
+  const currentVersion = getPackageJsonVersion()
+  // v1.2.3
+  const gitTag = `v${currentVersion}`
+
+  let result = null
+
+  if (releaseType === 'monthly') {
+    result = {
+      version: semver.inc(currentVersion, 'minor'),
```
Comment (m0ksem) [inline packages/ui/scripts/release-script.ts:59]:
```suggestion
      version: semver.inc(currentVersion, 'major'),
```

File: packages/ui/scripts/release-script.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      gitTag,
+      distTag: 'latest',
+      shouldCommit: true,
+      requiredBranch: 'master',
+    }
+  }
+  if (releaseType === 'weekly') {
+    result = {
+      version: semver.inc(currentVersion, 'minor'),
```
Comment (m0ksem) [inline packages/ui/scripts/release-script.ts:68]:
```suggestion
      version: semver.inc(currentVersion, 'patch'),
```

File: packages/ui/scripts/release-script.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+const argv: {
+  releaseType: ReleaseType,
+  semverReleaseType: string,
+  dryRun: boolean
+} = yargs(hideBin(process.argv))
+  .option('releaseType', {
+    description: 'Internal release type',
+    choices: ['monthly', 'weekly', 'next', 'experimental'],
```
Comment (m0ksem) [inline packages/ui/scripts/release-script.ts:34]:
```suggestion
    choices: ['major', 'minor', 'patch', 'next', 'experimental'],
```

How about to make it a big easier?

**Comment by m0ksem** [comment]
If we update vue-cli-plugin verison, we need to make sure we release vue-cli-plugin as well.

File: packages/bundlers-tests/package.json [hunk 6]
```diff
@@ -20,7 +20,6 @@
     "@vue/cli-service": "5",
     "jest": "^27.0.6",
     "rollup-plugin-visualizer": "^5.5.4",
-    "ts-node": "^10.1.0",
```
Comment (m0ksem) [inline packages/bundlers-tests/package.json:23]: ts-node is required for jest as per dependency.

File: packages/ui/scripts/release-script.ts [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+const getReleaseConfig = async (releaseType: ReleaseType): Promise<ReleaseConfig> => {
+  const currentBranchName = await executeCommand('git rev-parse --abbrev-ref HEAD')
+  // 20220602
+  const dateIsoShort = new Date().toISOString().split('T')[0].replaceAll('-', '')
+  // 12a4dd78
+  const commitHash = await getCommitHash()
+  // 1.2.3
+  const currentVersion = getPackageJsonVersion()
+  // v1.2.3
+  const gitTag = `v${currentVersion}`
```
Comment (m0ksem) [inline packages/ui/scripts/release-script.ts:53]: git tag needs to use bumped version, not current.

**Comment by m0ksem** [comment]
Build must execute before pushing...
image

HOPE BUILD WILL BE SUCCESED 😁
Waiting... 😖

File: packages/ui/scripts/release-script.ts (hunk 5 above)
Comment (asvae) [reply packages/ui/scripts/release-script.ts:34]: I don't think it's a good idea to keep semver release typing for vuestic release process which has nothing to do with semver.

**Comment by asvae** [comment]
Moved remaining issues to https://github.com/epicmaxco/vuestic-ui/issues/2004

**Pull request merged**